### PR TITLE
New: Add `crawler` tool

### PR DESCRIPTION
**Pull request by molant**
Fix sonarwhal/sonar#525

### CODE REVIEW

File: scripts/crawler/urls.txt [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+manoramaonline.com
+cricbuzz.com
+ntd.tv
+mmofreegames.online
+shopify.com
+chatwork.com
+cnzz.com
+ultimate-guitar.com
+wunderground.com
+olx.pl
```
Comment (alrra) [inline scripts/crawler/urls.txt]: Maybe add `tmz.com` ?

File: scripts/crawler/crawler.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+                if (!url.startsWith('http')) {
+                    url = `http://${entry}`;
+                }
+
+                new URL(url); // eslint-disable-line no-new
+
+                valids.push(url);
+            } catch (e) {
+                logger.error(`${entry} is not valid`);
```
Comment (alrra) [inline scripts/crawler/crawler.ts]: Note: `sonar` will automatically ignore invalid input and print the following message:

> Ignoring '\<input\>' as it's not an existing file nor a valid URL

File: scripts/crawler/crawler.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    totalSites = urls.length;
+};
+
+const run = async () => {
+    // read CLI to know what file to read
+    const start = Date.now();
+
+    try {
+        // `node crawler urlsFile`
+        logger.log('Reading list of urls');
```
Comment (alrra) [inline scripts/crawler/crawler.ts]: For consistency, can you use `URLs` instead if `urls` here and everywhere else? Thanks!

File: scripts/crawler/readme.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+1. Update the `.sonarrc` file to the desired configuration
+1. Update `urls.txt` with the urls to analyze
+1. Run the following command (assumes code is compiled and in the dist)
+   folder: `node crawler.js urls.txt`
+
+By default `crawler` will scan 10 URLs simultaneously. You can change
+that behavior via the command line. The following will scan 5 sites
+simultaneously `node crawler.js urls.txt 5`.
+
+You can also use `npm script crawl` to run the defaults.
```
Comment (alrra) [inline scripts/crawler/readme.md]: `crawl` script needs to be added in the `package.json` file.

File: scripts/crawler/urls.txt (hunk 1 above)
Comment (molant) [reply scripts/crawler/urls.txt]: I wasn't sure about the final list. This is just ~500 sites I copied from a list. Should we commit this list into the project?

File: scripts/crawler/crawler.ts (hunk 2 above)
Comment (molant) [reply scripts/crawler/crawler.ts]: Yes, but then we will spin another sonar instance and we will not know about the error until it tries to process that line. I've had problems in the past when I had to do a crawler for https://github.com/MicrosoftEdge/static-code-scan and failing after an hour or so

File: scripts/crawler/readme.md (hunk 4 above)
Comment (molant) [reply scripts/crawler/readme.md]: It's [already there](https://github.com/sonarwhal/tools/blob/master/package.json#L24) 😥

**Comment by molant** [comment]
@alrra all feedback should be addressed!

File: scripts/crawler/crawler.ts (hunk 2 above)
Comment (alrra) [reply scripts/crawler/crawler.ts]: Good to know! Thanks for the explanation.

File: scripts/crawler/urls.txt (hunk 1 above)
Comment (alrra) [reply scripts/crawler/urls.txt]: > I wasn't sure about the final list. This is just ~500 sites I copied from a list. Should we commit this list into the project?

Well, we should publish it, but we will a list Travis.

File: scripts/crawler/readme.md (hunk 4 above)
Comment (alrra) [reply scripts/crawler/readme.md]: Hmm, for some reason I didn't have `crawl` locally before.

File: scripts/crawler/urls.txt (hunk 1 above)
Comment (alrra) [reply scripts/crawler/urls.txt]: shouldn't*

**Pull request closed without merging**